app/views/questionnaire.py

```diff
@@ -464,6 +480,172 @@ def _render_template(context, current_location, template, previous_url, schema,
+
+
+class InvalidListItemId(Exception):
+    pass
+
+
+@with_list_store
+def validate_list_collector_route(list_store, schema, block_id, list_item_id=None):
+    block = schema.get_block(block_id)
+
+    if not block:
+        raise NotFound()
+
+    if block['type'] != 'ListCollector':
+        raise NotFound()
+
+    if list_item_id and list_item_id not in list_store[block['populates_list']]:
+        raise InvalidListItemId(f'List item id: {list_item_id} not found for block {block_id}')
+
+
+@questionnaire_blueprint.route('<block_id>/add', methods=['POST'])
+@login_required
+@with_answer_store
+@with_list_store
+@with_collection_metadata
+@with_metadata
+@with_schema
+def post_add_list_item(schema, metadata, collection_metadata, list_store, answer_store, block_id):
+
+    return list_collector_post_handler(schema, metadata, collection_metadata, list_store, answer_store, block_id,
+                                       None, 'add')
+
+
+@questionnaire_blueprint.route('<block_id>/<list_item_id>/remove', methods=['POST'])
+@login_required
+@with_answer_store
+@with_list_store
+@with_collection_metadata
+@with_metadata
+@with_schema
+def post_remove_list_item(schema, metadata, collection_metadata, list_store, answer_store, block_id,
+                          list_item_id):
+
+    return list_collector_post_handler(schema, metadata, collection_metadata, list_store, answer_store, block_id,
+                                       list_item_id, 'remove')
+
+
+@questionnaire_blueprint.route('<block_id>/<list_item_id>/edit', methods=['POST'])
+@login_required
+@with_answer_store
+@with_list_store
+@with_collection_metadata
+@with_metadata
+@with_schema
+def post_edit_list_item(schema, metadata, collection_metadata, list_store, answer_store, block_id,
+                        list_item_id):
+
+    return list_collector_post_handler(schema, metadata, collection_metadata, list_store, answer_store, block_id,
+                                       list_item_id, 'edit')
+
+
+@questionnaire_blueprint.route('<block_id>/<list_item_id>/remove', methods=['GET'])
+@login_required
+@with_answer_store
+@with_metadata
+@with_schema
+def get_remove_list_item(schema, metadata, answer_store, block_id, list_item_id):
+    return list_collector_get_handler(schema, metadata, answer_store, block_id, 'remove', list_item_id)
+
+
+@questionnaire_blueprint.route('<block_id>/<list_item_id>/edit', methods=['GET'])
+@login_required
+@with_answer_store
+@with_metadata
+@with_schema
+def get_edit_list_item(schema, metadata, answer_store, block_id, list_item_id):
+    return list_collector_get_handler(schema, metadata, answer_store, block_id, 'edit', list_item_id)
+
+
+@questionnaire_blueprint.route('<block_id>/add', methods=['GET'])
+@login_required
+@with_answer_store
+@with_metadata
+@with_schema
+def get_add_list_item(schema, metadata, answer_store, block_id):
+    return list_collector_get_handler(schema, metadata, answer_store, block_id, 'add', '')
```

Comment:
The `list_item_id` of `''` here needs to be addressed. Issue being that currently, the None value represents 'do not filter on list_item_id'.